atom/browser/ui/cocoa/atom_menu_controller.mm

```diff
@@ -139,6 +152,28 @@ - (void)addItemToMenu:(NSMenu*)menu
             fromModel:(atom::AtomMenuModel*)model {
   base::string16 label16 = model->GetLabelAt(index);
   NSString* label = l10n_util::FixUpWindowsStyleLabel(label16);
+  base::string16 role = model->GetRoleAt(index);
+
+  if (role == base::ASCIIToUTF16("openrecent")) {
+    // Remove singleton menu item from parent menu
+    [[openRecentMenuItem_ menu] removeItem:openRecentMenuItem_];
```

Comment:
Not sure how the relationship between `NSMenuItem` and `NSMenu` is facilitated. 😕  Since we are retaining the `NSMenuItem` the first time setting a template menu, the menu item should persist throughout its lifetime. But would its parent `NSMenu` remove its relationship from the menu item when it's deallocated?

Reply:
In response to this... Never assign the `menu` for `NSMenuItem` nor the `supermenu` for `NSMenu`.